JEDI-Flipfit-JAVA-POS/src/com/flipkart/dao/daoImpl/GymDAOImpl.java

```diff
@@ -3,12 +3,91 @@
  */
 package com.flipkart.dao.daoImpl;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import com.flipkart.bean.Gym;
 import com.flipkart.dao.GymDAO;
 
 /**
  * 
  */
 public class GymDAOImpl implements GymDAO {
+private static GymDAO gymDAOObj = null;
+	
+	private GymDAOImpl() {
+		
+	}
+	
+	public static synchronized GymDAO
+	getInstance() {
+		if (gymDAOObj == null)
+			gymDAOObj = new GymDAOImpl();
+
+		return gymDAOObj;
+	}
+	
+	private ArrayList<Gym> gyms = new ArrayList<Gym>();
+	private int gymCount = 0;
+
+	@Override
+	public List<Gym> getAllGyms() {
+		return gyms;
+	}
+
+	@Override
+	public Gym getGym(int gymId) {
+		for(Gym g: gyms) {
+			if(g.getGymId() == gymId) return g;
+		}
+		System.out.println("Gym not found");
+		return null;
+	}
+
+	@Override
+	public List<Gym> pendingGymRequest() {
+		List<Gym> pendingGyms =  new ArrayList<Gym>();
+		for(Gym g: gyms) {
+			if(!g.isApproved())  pendingGyms.add(g);
+		}
+		return pendingGyms;
+	}
+
+	@Override
+	public void addGym(Gym gym) {
+		gym.setGymId(gymCount++);
+		gyms.add(gym);
+	}
+
+
+
+	@Override
+	public void removeGym(int gymId) {
+		for(Gym g: gyms) {
+			if(g.getGymId() == gymId) { 
+				g.setActive(false);
```

Comment:
WILL DO LATER: when gym is removed then we should remove all the bookings and available slots of that gym.